### PR TITLE
Feature/edit user profile

### PR DESCRIPTION
**Pull request by Manbeartrain**

### CODE REVIEW

File: src/screens/LandingScreen.js [hunk 1]
```diff
@@ -12,12 +12,16 @@ import { COLOR_PALETTE, NAVIGATORS } from "../utils/constants";
...
 import { setPlaylists } from "../redux/MusicSlice";
+import { getFirebaseUser } from "../utils/firebase";
 
 const LandingScreen = ({ navigation }) => {
 	const dispatch = useDispatch();
+	const user = getFirebaseUser();
 
 	const loadInitialPlaylists = async () => {
-		await auth().signInAnonymously();
+		if (!user) {
```
Comment (daboigbae) [inline src/screens/LandingScreen.js:22]: You don't need to check for authenticaiton here. Always grab the playlist data using anon, and then profile updates with the signed in user

File: src/screens/userProfileScreens/EditUserProfileScreen.js [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+	Dimensions,
+	PixelRatio,
+	SafeAreaView,
+	ScrollView,
+	StyleSheet,
+	TextInput,
+	View
+} from "react-native";
+import PropTypes from "prop-types";
+import React, { useMemo, useState } from "react";
```
Comment (daboigbae) [inline src/screens/userProfileScreens/EditUserProfileScreen.js:12]: should be the first import

File: src/screens/userProfileScreens/EditUserProfileScreen.js [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+import UserProfileAvatar from "../../components/profile/UserProfileAvatar";
+import { useDispatch, useSelector } from "react-redux";
```
Comment (daboigbae) [inline src/screens/userProfileScreens/EditUserProfileScreen.js:14]: remember our imports should come after library imports

File: src/screens/userProfileScreens/EditUserProfileScreen.js [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+	);
+
+	const photoURL = useSelector(
+		({ UserSlice }) => UserSlice?.userInformation?.photoURL
+	);
+
+	const [editUserName, setEditUsername] = useState(
+		displayName || "Default User"
+	);
+	const [imagePreview, setImagePreview] = useState(photoURL || "");
```
Comment (daboigbae) [inline src/screens/userProfileScreens/EditUserProfileScreen.js:39]: why does this have to be photoURL or empty string?

File: src/screens/userProfileScreens/EditUserProfileScreen.js [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+	const [isLoading, setIsLoading] = useState(false);
+
+	const informationNotUpdated = useMemo(() => {
+		return displayName === editUserName && photoURL === imagePreview;
+	});
```
Comment (daboigbae) [inline src/screens/userProfileScreens/EditUserProfileScreen.js:44]: you need to add a dependency array to this

File: src/screens/userProfileScreens/EditUserProfileScreen.js [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+
+	const handleSelectImage = async () => {
```
Comment (daboigbae) [inline src/screens/userProfileScreens/EditUserProfileScreen.js:46]: let's move this to a images util

Comment (daboigbae) [reply src/screens/userProfileScreens/EditUserProfileScreen.js:46]: something like handleSelectImage(callback) where you can set-up the callback from where it's called

**Review by daboigbae — CHANGES_REQUESTED** [review]

**Review by daboigbae — CHANGES_REQUESTED** [review]
A few things here and then let's make the edit screen come up from the bottom exactly how spotify does theirs. 

Allow the user to only change image and/or username. Account for empty images, not for empty usernames

Add padding to the right of the account icon

File: src/screens/userProfileScreens/EditUserProfileScreen.js (hunk 4 above)
Comment (Manbeartrain) [reply src/screens/userProfileScreens/EditUserProfileScreen.js:39]: When the user first creates account everything is set to default. So If there is no photoURL available (meaining they haven't already updatred their profile once) then the image will show the empty image icon instead

**Review by daboigbae — APPROVED** [review]

**Pull request merged**